likecoin/public/likecoin.php

```diff
@@ -25,6 +25,61 @@
  */
 require_once dirname( __FILE__ ) . '/likecoin-button.php';
 require_once dirname( __FILE__ ) . '/web-monetization.php';
+require_once dirname( __FILE__ ) . '/iscn-badge.php';
+
+/**
+ * Add LikeButton to post content if suitable
+ *
+ * @param string| $content The original post content.
+ */
+function likecoin_content_filter( $content ) {
+	global $post;
+	$option          = get_option( LC_BUTTON_OPTION_NAME );
+	$post_type_query = '';
+
+	if ( ! isset( $post ) ) {
+		return $content;
+	}
+	do {
+		// follow post meta if option is not set yet.
+		if ( ! isset( $option[ LC_OPTION_BUTTON_DISPLAY_AUTHOR_OVERRIDE ] ) || $option[ LC_OPTION_BUTTON_DISPLAY_AUTHOR_OVERRIDE ] ) {
+			$widget_option   = get_post_meta( $post->ID, LC_OPTION_WIDGET_OPTION, true );
+			$widget_position = isset( $widget_option[ LC_OPTION_WIDGET_POSITION ] ) ? $widget_option[ LC_OPTION_WIDGET_POSITION ] : '';
+			if ( strlen( $widget_position ) > 0 ) {
+				if ( 'none' === $widget_position ) {
+					// set to none, exit early.
+					return $content;
+				}
+				// else if set post_meta, cont to render.
+				break;
+			}
+			// otherwise judge by switch case below.
+		}
+		if ( isset( $option[ LC_OPTION_BUTTON_DISPLAY_OPTION ] ) ) {
+			$type = $option[ LC_OPTION_BUTTON_DISPLAY_OPTION ];
+			switch ( $type ) {
+				case 'none':
+					// set to none, exit early.
+					return $content;
+				case 'post':
+					$post_type_query = 'post';
+					break;
+				case 'always':
+					// no op.
+					break;
+			}
+		}
+		// default show LikeButton if id exist.
+	} while ( false );
```

Comment:
These parts are LikeCoin Button settings, and should not affect showing ISCN badge or not. Currently since these part call `return $content`, this will avoid ISCN badge for being shown